test/flows/benefit_cap_calculator_flow_test.rb

```diff
@@ -0,0 +1,311 @@
+    end
+
+    should "render question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      context "when the postcode is in London" do
+        setup do
+          @postcode = "WC2B 6SE"
+          stub_postcode(@postcode, region: "London")
+        end
+
+        should "have a next node of outcome_affected_greater_than_cap_london for benefits greater than the cap" do
+          cap = benefit_cap(region: :london)
+          add_responses housing_benefit_amount?: (cap + 100).to_s
+          assert_next_node :outcome_affected_greater_than_cap_london, for_response: @postcode
+        end
+
+        should "have a next node of outcome_not_affected_less_than_cap_london for benefits greater than the cap" do
+          assert_next_node :outcome_not_affected_less_than_cap_london, for_response: @postcode
+        end
+      end
+
+      context "when the postcode is outside London" do
+        setup do
+          @postcode = "B1 1PW"
+          stub_postcode(@postcode, region: "West Midlands")
+        end
+
+        should "have a next node of outcome_affected_greater_than_cap_national for benefits greater than the cap" do
+          cap = benefit_cap(region: :national)
+
+          add_responses housing_benefit_amount?: (cap + 100).to_s
+          assert_next_node :outcome_affected_greater_than_cap_national, for_response: @postcode
+        end
+
+        should "have a next node of outcome_not_affected_less_than_cap_national for benefits greater than the cap" do
+          assert_next_node :outcome_not_affected_less_than_cap_national, for_response: @postcode
+        end
+      end
+    end
+  end
+
+  context "outcome: outcome_affected_greater_than_cap_london" do
+    setup do
+      testing_node :outcome_affected_greater_than_cap_london
+      stub_postcode("WC2B 6SE", region: "London")
+      add_responses receive_housing_benefit?: "yes",
+                    working_tax_credit?: "no",
+                    receiving_exemption_benefits?: [],
+                    receiving_non_exemption_benefits?: %w[jsa],
+                    single_couple_lone_parent?: "single",
+                    enter_postcode?: "WC2B 6SE"
+    end
+
+    should "render new housing benefit amount by subtracting housing benefit from other benefits" do
+      cap = benefit_cap(region: :london)
+      add_responses jsa_amount?: (cap - 20).to_s,
+                    housing_benefit_amount?: cap.to_s
+      assert_rendered_outcome text: "Your Housing Benefit will be: £20"
+    end
+
+    should "reduce housing benefit amount to a minimum of 50p" do
+      cap = benefit_cap(region: :london)
+      add_responses jsa_amount?: cap.to_s,
+                    housing_benefit_amount?: cap.to_s
+      assert_rendered_outcome
+      assert_match "Your Housing Benefit will be: 50p", @test_flow.outcome_body_text
+      assert_match "Your Housing Benefit won’t be reduced to less than 50p", @test_flow.outcome_body_text
```

Comment:
It's great that these values are being tested, especially as I can't see that they're being tested anywhere else.

Is that the case for the other values. e.g. `total_over_cap`. Is that value being tested anywhere else? If not, perhaps it and the others could be tested for this outcome? I wouldn't be expect them to be duplicated in the other test. 

From a quick look, I think there are 4 values in total that are calculated in the flow for use in the outcome:

`benefit_cap`
`housing_benefit_amount`
`total_benefits_amount`
`total_over_cap`
 
It would be great if one of the lines that each appears on could be tested.